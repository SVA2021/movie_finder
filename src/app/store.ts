import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import authReducer from '../features/auth/authSlice';
import movieFinderReducer from '../features/movieFinder/movieFinderSlice';

export const store = configureStore({
  reducer: {
    auth: authReducer,
		movieFinder: movieFinderReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
