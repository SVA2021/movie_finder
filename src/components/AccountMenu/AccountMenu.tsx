import {AccountCircle, Logout, Settings} from '@mui/icons-material';
import {Grid, Popover} from '@mui/material';
import * as React from 'react';
import {StyledButton, StyledTextItem,} from '../../components'

export const AccountMenu = () => {
	const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
	const open = Boolean(anchorEl);
	const handleClick = (event: React.MouseEvent<HTMLElement>) => {
		setAnchorEl(event.currentTarget);
	};
	const handleClose = () => {
		setAnchorEl(null);
	};

	return (
		<>
			<StyledButton
				id="account-button"
				aria-controls={open ? 'account-menu' : undefined}
				aria-haspopup="true"
				aria-expanded={open ? 'true' : undefined}
				onClick={handleClick}
				fullWidth
			>
				<StyledTextItem >Login<AccountCircle fontSize='large' /></StyledTextItem>
			</StyledButton>
			<Popover
				id="main-menu"
				anchorEl={anchorEl}
				open={open}
				onClose={handleClose}
				anchorOrigin={{
					vertical: 'bottom',
					horizontal: 'center',
				}}
				transformOrigin={{
					vertical: 'top',
					horizontal: 'right',
				}}
			>
				<Grid container direction={'column'} spacing={2} sx={{
					bgcolor: 'primary.main',
					p: 2,
				}} >
					<Grid item xs={12} md={6} lg={3}>
						<StyledTextItem alignItems={'flexEnd'} ><AccountCircle fontSize='medium' />Account</StyledTextItem>
					</Grid>
					<Grid item xs={12} md={6} lg={3}>
						<StyledTextItem ><Settings fontSize='medium' />Settings</StyledTextItem>
					</Grid>
					<Grid item xs={12} md={6} lg={3}>
						<StyledTextItem ><Logout fontSize='medium' />Logout</StyledTextItem>
					</Grid>
				</Grid>
			</Popover>
		</>
	);
}
