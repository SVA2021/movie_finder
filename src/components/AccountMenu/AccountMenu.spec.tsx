import {render, screen, fireEvent} from "@testing-library/react";
import {BrowserRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import {store} from "../../app/store";
import {AccountMenu} from "./AccountMenu";

describe('render AccountMenu', () => {

  const setUp = () => render(
    <Provider store={store}>
      <BrowserRouter>
        <AccountMenu />
      </BrowserRouter>
    </Provider>
  );

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('account-menu')).toMatchSnapshot();
  });

  it('check button open menu', () => {
    setUp();
    fireEvent.click(screen.getByTestId('panel-account'))
    expect(screen.getByTestId('panel-account-details')).toBeInTheDocument();
  });

})