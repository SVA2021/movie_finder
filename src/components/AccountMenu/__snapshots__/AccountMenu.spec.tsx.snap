// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AccountMenu snapshot test 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
  data-testid="account-menu"
>
  <div
    aria-controls="panel-account"
    aria-expanded="false"
    class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
    data-testid="panel-account"
    id="panel-account"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
    >
      <p
        class="MuiTypography-root MuiTypography-h3 css-14at1vf-MuiTypography-root"
        data-testid="styled-text-item"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AccountCircleIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
          />
        </svg>
        Аккаунт
      </p>
    </div>
    <div
      class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ExpandMoreIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </div>
  </div>
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          aria-labelledby="panel-account"
          class="MuiAccordion-region"
          id="panel-account"
          role="region"
        >
          <div
            class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
            data-testid="panel-account-details"
          >
            <form
              class="MuiBox-root css-0"
              data-testid="account-form"
            >
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-acv73g-MuiInputBase-root-MuiInput-root"
                required=""
              >
                <input
                  class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  name="login"
                  placeholder="логин"
                  required=""
                  type="text"
                  value=""
                />
              </div>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-acv73g-MuiInputBase-root-MuiInput-root"
                required=""
              >
                <input
                  class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  name="password"
                  placeholder="пароль"
                  required=""
                  type="text"
                  value=""
                />
              </div>
              <div
                class="MuiBox-root css-16j30i3"
              >
                <button
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="submit"
                >
                  Войти
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
